lib/netutil/netutil.go

```diff
@@ -16,3 +23,22 @@ func AddressURL(network, host string) string {
 	}
 	return u.String()
 }
+
+// Gateway returns the IP address of the default network gateway.
+func Gateway() (ip net.IP, err error) {
+	ip, err = gateway.DiscoverGateway()
+	if err != nil {
+		// Fails on Android 14+ due to permission denied error when reading
+		// /proc/net/route. The wrapper may give a hint then because it is
+		// able to discover the gateway from java code.
+		if v := os.Getenv("ANDROID_NET_GATEWAY_IPV4"); v != "" {
```

Comment:
I'd drop `ANDROID` from the name, could be used on any system. And probably that's not even a hypothetical, I'd be surprised if iOS allowed this kind of access. I'd go with `FALLBACK_NET_GATEWAY_IPV4`.